### PR TITLE
PRfD - PR Details page

### LINKED ISSUE #8175: PRfD - WebUI - View PR
Add a page for viewing a single PR.

This includes:
TBD.

### PR DESCRIPTION
**Pull request by itaigilo**
Closes #8175 

## Change Description

Create the page showing the PR Details.

Also handles the navigation, and creates a scaffold fo the Create PR page (as part of it).

Note: the PR's description is currently simple test. Will change it to md viewer when I'll implement the PR edit. 

-----

PR Details

### CODE REVIEW

File: webui/src/lib/api/index.js [hunk 1]
```diff
@@ -634,13 +657,13 @@ export const uploadWithProgress = (url, file, method = 'POST', onProgress = null
...
         });
         xhr.addEventListener('load', () => {
-          resolve({
-              status: xhr.status,
-              body: xhr.responseText,
-              contentType: xhr.getResponseHeader('Content-Type'),
-              etag: xhr.getResponseHeader('ETag'),
-              contentMD5: xhr.getResponseHeader('Content-MD5'),
-          })
+            resolve({
```
Comment (itaigilo) [inline webui/src/lib/api/index.js:660]: Accidentally formatted this. White spaces only from here in this file.

File: webui/src/constants.ts [hunk 2]
```diff
@@ -1,22 +1,16 @@
...
-    Changed = "changed",
-}
+
 export enum TreeRowType {
     Object,
     Prefix,
     Table
 }
 
+export enum PullStatus {
```
Comment (itaigilo) [inline webui/src/constants.ts:11]: This seems like a better place for this.

**Review by N-o-Z — CHANGES_REQUESTED** [review]
It looks very very nice.
The only reason I'm requesting changes is that there seems to be some mix up with the PR titles and branch src-dest when going from lists to the actual PR.
This is how it looks in the list:
image
And this is how it looks when I press on the PR (notice the title, src and dest branches):
image

**Comment by N-o-Z** [comment]
@itaigilo Since we already merged Get/Create E2E you can actually implement it now.
Also, I don't see a reason not to implement the Merge logic already since this is unrelated to the PR logic

**Comment by itaigilo** [comment]
> The only reason I'm requesting changes is that there seems to be some mix up with the PR titles and branch src-dest when going from lists to the actual PR. 

@N-o-Z If I understand you correctly - the difference is because of the mock data (which is different between the GET and LIST API calls).

And regarding your comment about integrating first - 
I'd be happy to, but currently I have no real PRs data to test the integration of the Details page.
My plan is to merge this PR, continue to creating the Create page (which will be connected to the API and will create a real PR in the KV), and then integrate the rest.

**Comment by N-o-Z** [comment]
> > The only reason I'm requesting changes is that there seems to be some mix up with the PR titles and branch src-dest when going from lists to the actual PR.
> 
> @N-o-Z If I understand you correctly - the difference is because of the mock data (which is different between the GET and LIST API calls).
> 
> And regarding your comment about integrating first - I'd be happy to, but currently I have no real PRs data to test the integration of the Details page. My plan is to merge this PR, continue to creating the Create page (which will be connected to the API and will create a real PR in the KV), and then integrate the rest.

Thanks - I guess I missed that!

**Review by N-o-Z — APPROVED** [review]
LGTM!

**Pull request merged**